src/analysis/normed_space/star/complex.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2022 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.star.basic
+import algebra.star.self_adjoint
+import analysis.complex.basic
+
+/-!
+# Complex normed star modules and algebras
+
+Facts about star modules and star algebras over the complex numbers.
+
+## Main definitions
+
+* `star_ring.re`: the real part of an element of a star ring, defined as `2⁻¹ • (x + star x)`
+* `star_ring.im`: the imaginary part of an element of a star ring, defined as
+  `(-I * 2⁻¹) • (x - star x)`.
+
+-/
+
+variables {E : Type*}
+
+namespace star_ring
+open_locale complex_conjugate
+open complex
+
+variables [semi_normed_ring E] [star_add_monoid E] [module ℂ E] [star_module ℂ E]
+
+/-- The real part of an element of star algebra. -/
+@[simps] noncomputable def re (x : E) : self_adjoint E :=
```

Comment:
I was going to suggest instead defining:
```suggestion
@[simps] noncomputable def re : E →ₗ[ℝ] self_adjoint E :=
```
until I realised that we don't have `module ℝ (self_adjoint E)`.

Would it be easy to add that instance as part of this PR and implement the suggestion?

(Similar remarks for `im` of course.)

Reply:
Good point! Yes we should definitely have that instance, I'll try to get to it soon.